### ISSUE TITLE
Eslint doesn't work with error Unknown options: dotfiles

### ISSUE THREAD

**Issue by machsix** [microsoft/vscode-eslint#1747]
Here is my github repo https://github.com/machsix/Super-preloader
Here is the error information
```
[Info  - 11:59:17 PM] ESLint server is starting.
[Info  - 11:59:17 PM] ESLint server running in node v18.15.0
[Info  - 11:59:17 PM] ESLint server is running.
[Info  - 11:59:21 PM] ESLint library loaded from: D:\Git\test2\node_modules\eslint\lib\api.js
[Error - 11:59:21 PM] An unexpected error occurred:
[Error - 11:59:21 PM] Error: Invalid Options:
- Unknown options: dotfiles
    at processOptions (D:\Git\test2\node_modules\eslint\lib\eslint\eslint.js:282:15)
    at new ESLint (D:\Git\test2\node_modules\eslint\lib\eslint\eslint.js:429:34)
    at q (c:\Users\xx\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:18735)
    at E (c:\Users\xx\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:19173)
    at e.validate (c:\Users\xx\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:24819)
    at c:\Users\xx\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:221494
```

Here are vscode info
```
Version: 1.84.2 (user setup)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:51:52.184Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
```

I'm using v2.4.2 version of the extension

**Comment by machsix** [comment]
If I run eslint directly from CLI, I don't see any error.

**Comment by machsix** [comment]
Hi Dirk, I created a more simple repo here https://github.com/machsix/eslint_test but still observed the error.

I also completely uninstalled Visual Studio Code, removed all configurations under ~/.vscode, and reinstalled the latest version
```
Version: 1.85.1 (user setup)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:49:37.021Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045
```
I only installed one extension `Eslint`. Here is the error from Output->Eslint
```
[Info  - 11:43:59 PM] ESLint server is starting.
[Info  - 11:43:59 PM] ESLint server running in node v18.15.0
[Info  - 11:43:59 PM] ESLint server is running.
[Info  - 11:44:00 PM] ESLint library loaded from: D:\Git\test\node_modules\eslint\lib\api.js
[Error - 11:44:00 PM] An unexpected error occurred:
[Error - 11:44:00 PM] Error: Invalid Options:
- Unknown options: dotfiles
    at processOptions (D:\Git\test\node_modules\eslint\lib\eslint\eslint.js:282:15)
    at new ESLint (D:\Git\test\node_modules\eslint\lib\eslint\eslint.js:429:34)
    at q (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:18735)
    at E (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:19173)
    at e.validate (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:24819)
    at c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:221494
[Error - 11:44:25 PM] An unexpected error occurred:
[Error - 11:44:25 PM] Error: Invalid Options:
- Unknown options: dotfiles
    at processOptions (D:\Git\test\node_modules\eslint\lib\eslint\eslint.js:282:15)
    at new ESLint (D:\Git\test\node_modules\eslint\lib\eslint\eslint.js:429:34)
    at q (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:18735)
    at E (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:19173)
    at e.validate (c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:24819)
    at c:\Users\x\.vscode\extensions\dbaeumer.vscode-eslint-2.4.2\server\out\eslintServer.js:1:221494
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```
Here is my node information
```
PS D:\Git\test> node --version
v20.10.0
```

**Comment by machsix** [comment]
An update:
I completely deleted folder `C:\Users\x\AppData\Roaming\Code` and now the extension works. Not sure what happened...
